plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,538 @@
+
+                    elif not attribute_is_empty and not match.details.get(attribute_name):
+                        update_mso_values_attributes_payload(mso_values, {attribute_name: attribute_dict})
+
+                    elif not attribute_is_empty and match.details.get(attribute_name):
+                        mso_values[(attribute_name, "enabled")] = attribute_dict.get("enable")
+                        if attribute_name == "ospf":
+                            mso_values[(attribute_name, "authType")] = attribute_dict.get("authentication_type")
+                        else:
+                            mso_values[(attribute_name, "authEnabled")] = attribute_dict.get("authentication")
+                        mso_values[(attribute_name, "keyID")] = attribute_dict.get("key_id")
+
+                        if attribute_dict.get("key") is not None:
+                            if match.details.get(attribute_name, {}).get("key") is not None:
+                                mso_values[(attribute_name, "key", "value")] = attribute_dict.get("key")
+                            else:
+                                mso_values[(attribute_name, "key")] = dict(value=attribute_dict.get("key"))
+
+            append_update_ops_data(ops, match.details, interface_group_policy_path, mso_values, mso_values_remove)
+            mso.sanitize(match.details, collate=True)
+        else:
+            update_mso_values_attributes_payload(mso_values, attributes_dict)
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path=interface_group_policy_path, value=mso_values))
+
+    elif state == "absent":
+        if mso.existing:
+            ops.append(dict(op="remove", path=interface_group_policy_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        l3outs = response.get("l3outTemplate", {}).get("l3outs", [])
+        l3out_object = mso_template.get_object_by_key_value_pairs(
+            "L3Out",
+            l3outs,
+            [KVPair("uuid", l3out_identifier.get("uuid")) if l3out_identifier.get("uuid") else KVPair("name", l3out_identifier.get("name"))],
+            True,
+        )
+        interface_groups = l3out_object.details.get("interfaceGroups", [])
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            interface_groups,
+            [KVPair("name", name)],
+        )
+        if match:
+            for attribute_name in attributes_dict.keys():
+                if match.details.get(attribute_name, {}).get("key", {}).get("ref"):
+                    match.details[attribute_name]["key"].pop("ref", None)
+            mso.existing = match.details
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        mso.existing = mso.proposed if state == "present" else {}
```

Comment:
Insert the `Interface Routing Policy` and `Custom QoS Policy` object names in the `current` and `previous` portion of the module output when we do `create, update, query and delete`.